lib/player.rb

```diff
@@ -0,0 +1,18 @@
+class Player
+  attr_accessor :deck, :has_lost
+  attr_reader :name
+  
+  def initialize(name, deck)
+    @name = name
+    @deck = deck
+    @has_lost = false
+  end
+  
+  def has_lost?
+    if @deck.cards.count.zero?
+      @has_lost = true
+    else
+      @has_lost
+    end
```

Comment:
This feels a little complicated. I'd get rid of the `@has_lost` (and associated `attr_reader`) and just return `true` or `false` directly here.